crates/wysiwyg/src/dom/dom_handle.rs

```diff
@@ -14,61 +14,82 @@
 
 #[derive(Clone, Debug, PartialEq)]
 pub struct DomHandle {
-    // Later, we will want to allow continuing iterating from this handle, and
-    // comparing handles to see which is first in the iteration order. This
-    // will allow us to walk the tree from earliest to latest of 2 handles.
-    path: Vec<usize>,
+    // The location of a node in the tree, or None if we don't know yet
+    path: Option<Vec<usize>>,
 }
 
 impl DomHandle {
+    /// Create a new handle with the provided path.
+    /// So long as the path provided points to a valid node, this handle
+    /// can be used (it is set).
     pub fn from_raw(path: Vec<usize>) -> Self {
-        Self { path }
+        Self { path: Some(path) }
     }
 
-    pub fn parent_handle(&self) -> DomHandle {
-        assert!(self.path.len() > 0);
+    /// Create a new UNSET handle
+    /// Don't use this handle for anything until it has been set.
+    /// Most methods will panic!
+    pub fn new_unset() -> Self {
+        Self { path: None }
+    }
 
-        let mut new_path = self.path.clone();
-        new_path.pop();
-        DomHandle::from_raw(new_path)
+    /// Returns true if this handle has been set to a value
+    pub fn is_set(&self) -> bool {
+        !self.path.is_none()
+    }
+
+    /// Returns true if this handle refers to a root node
+    /// Panics if this handle is unset.
+    pub fn is_root(&self) -> bool {
+        let path = self.path.as_ref().expect("Handle is unset!");
```

Comment:
Since this line is repeated in most functions, could we maybe create a small private function to replace it? So instead of:

```
let path = self.path.as_ref().expect("Handle is unset!");
path.is_empty()
```

We could write something like:

```
self.safe_path().is_empty()
```

Reply:
Ha, I did this and then realised the function already existed! (`raw`)